docs/docsite/rst/developing_modules_general_aci.rst

```diff
@@ -0,0 +1,1851 @@
+.. _aci_dev_guide:
+
+****************************
+Developing Cisco ACI modules
+****************************
+This is a developer guide for contributing modules to the Cisco ACI-Ansible collection. It is for developers who want to contribute code, fixes, or new modules to the collection. It will walk you through different steps and the expected conventions for contributions.
+
+For more information about Cisco ACI, look at the `Cisco ACI user guide <https://www.cisco.com/c/en/us/solutions/collateral/data-center-virtualization/application-centric-infrastructure/solution-overview-c22-741487.html>`_.
+
+What's covered in this section:
+
+.. contents::
+  :depth: 3
+  :local:
+
+.. _aci_dev_guide_intro:
+
+Introduction
+============
+The `cisco.aci collection <https://galaxy.ansible.com/cisco/aci>`_ already includes a large number of Cisco ACI modules; however, the ACI object model is huge, and covering all possible functionality would easily require more than 1,500 individual modules. Therefore, Cisco develops modules requested by people on a just-in-time basis.
+
+If you need a specific functionality, you have 3 options:
+
+- Open an issue using https://github.com/CiscoDevNet/ansible-aci/issues/new/choose so that Cisco developers can build, enhance, or fix the modules for you.
+- Learn the ACI object model and use the low-level APIC REST API using the `aci_rest <https://docs.ansible.com/ansible/latest/collections/cisco/aci/aci_rest_module.html>`_ module.
+- Contribute to Cisco's ansible-aci project by writing your own dedicated modules, proposing a fix or an enhancement, and being part of the Cisco ansible-aci community.
+
+.. _aci_dev_guide_git:
+
+In this guide, we're going to concentrate on the third option to show you how to build your own module, fix an issue, or improve an existing module and contribute it back to the ansible-aci project. The first step in the process is to retrieve the latest version of the ansible-aci collection code.
+By retrieving the latest version, you will be able to modify existing code.
+
+Fork, Clone and Branch
```

Comment:
I am not sure if I agree with the sections, from preview. 

- Intro
- Fork/Clone/Branch
- Module Structure
   - the rest of all the chapters

To me it makes more sense to structure it differently for example something like:

- Introduction
- Development environment
   - Git specific things like forking etc
   - Tools required etc
   - Coding conventions etc
- Collection Details
   - Explanation of module
   - Explanation of utils
   - Explanation of constants
- Testing
   - Sanity
   - Integration
   - Coverage